presto-spi/src/main/java/com/facebook/presto/spi/plan/ValuesNode.java

```diff
@@ -37,13 +37,15 @@
 {
     private final List<VariableReferenceExpression> outputVariables;
     private final List<List<RowExpression>> rows;
+    private final Optional<String> origTableName;
```

Comment:
Just call it label or something and also maybe use the alias that the use gives to values nodes as the label for other situations.

Reply:
Can you also give a warning (if you are not already) when a table is eliminated?

Reply:
> Can you also give a warning (if you are not already) when a table is eliminated?

Added warning in HiveFilterPushDown, it will output warnings like

> WARNING: Table 'TableHandle {connectorId='hive', connectorHandle='HiveTableHandle{schemaName=tpch, tableName=nation, analyzePartitionValues=Optional.empty}', layout='Optional[tpch.nation{domains={nationkey=[ [["100"]] ]}}]'}' is converted to ValuesNode by HiveFilterPushdown

> Just call it label 

Rename it to valuesNodeLabel

> also maybe use the alias that the use gives to values nodes as the label for other situations.

Looking into it now.

Reply:
> Added warning in HiveFilterPushDown, it will output warnings like
> 
> > WARNING: Table 'TableHandle {connectorId='hive', connectorHandle='HiveTableHandle{schemaName=tpch, tableName=nation, analyzePartitionValues=Optional.empty}', layout='Optional[tpch.nation{domains={nationkey=[ [["100"]] ]}}]'}' is converted to ValuesNode by HiveFilterPushdown
> 

ValueNode is a plan concept. So make it clear to the user what it did exactly. I think we use it only the partition is empty. So make that explicit and say:

table ... returned 0 rows

or something like that

Reply:
> ValueNode is a plan concept. So make it clear to the user what it did exactly. I think we use it only the partition is empty. So make that explicit and say:
>
> table ... returned 0 rows

Changed the warning to
> WARNING: Table 'TableHandle {connectorId='hive', connectorHandle='HiveTableHandle{schemaName=tpch, tableName=nation, analyzePartitionValues=Optional.empty}', layout='Optional[tpch.nation{domains={nationkey=[ [["100"]] ]}}]'}' **returns 0 rows, and is converted to an empty ValuesNode** by HiveFilterPushdown

> also maybe use the alias that the use gives to values nodes as the label for other situations.

I checked the code, the alias information is not directly available when creating the ValuesNode, and needs non trival changes to the call stack to pass this information. Consider that this is a good to have change and not directly related to the problem fixed in the pull request. Could we leave it as it is for now? thanks!

Reply:
Looks good. Please add a separate task for propagating the values node alias to plan